lib/collection/src/shards/shard_holder.rs

```diff
@@ -412,8 +505,12 @@ impl ShardHolder {
                 debug_assert!(false, "Do not expect empty shard selector")
             }
             ShardSelectorInternal::All => {
-                for (shard_id, shard) in self.shards.iter() {
-                    let shard_key = self.shard_id_to_key_mapping.get(shard_id);
+                for (&shard_id, shard) in self.shards.iter() {
+                    if self.resharding == Some(shard_id) {
+                        continue;
+                    }
```

Comment:
Ignore resharding shard when doing read operations. Otherwise, once you initialized the new shard, all read requests would be broken, because we did not fix read operations to work with resharding yet.

Reply:
Instead I'd prefer to depend on the `Resharding` replica set state. I feel like that more closely matches the periods we do and do not want to read from the new shard.

Reply:
I don't think that's practical (or even possible).

Also, this is pure WIP code to disable querying resharded shard, so that read operations still work while we implement proper read operation handling.

Reply:
Technically, yes, but `shard` (that is in scope in the loop) is a `ShardReplicaSet`.

So to query replica set state you have to reach into `ShardReplicaSet::replica_state` (which is `SaveOnDisk`, so you have to read-lock it first), and then you have to either specify `peer_id` (and I don't think we actually have one here to specify) or you have to loop over the whole replica set state hashmap.

Reply:
When doing resharding you *should* have only a single replica in the replica set. And this single replica *should* be in `Resharding` state (at least, it is for now). But it does seem a lot easier to just duplicate this info for now.

Maybe we come up with a better understanding what exactly should be "the source of truth" for resharding state later, I vote we leave this as-is for now.